install/certmonger/dogtag-ipa-ca-renew-agent-submit

```diff
@@ -65,8 +65,38 @@ if six.PY3:
 IPA_CA_NICKNAME = 'caSigningCert cert-pki-ca'
 
 def get_nickname():
-    csr = os.environ.get('CERTMONGER_CSR')
-    return pkcs10.get_friendlyname(csr) if csr else None
+    subject = os.environ.get('CERTMONGER_REQ_SUBJECT')
+    if not subject:
+        return None
+
+    subject_base = dsinstance.DsInstance().find_subject_base()
+    if not subject_base:
+        return None
+
+    nickname_by_subject_dn = {
```

Comment:
@jcholast right, but if we get certmonger to put the cert location (which for NSSDB, includes nickname) into the helper environment, the subject does not need to be referenced at all, to find the nickname.

Recall that we used to read nickname directly from PKCS #10 "friendlyName" attribute, but that was removed to avoid doing that parsing work ourselves.